minvws/nl-kat-coordination
# Fix scheduled Aggregate Report naming

**NOTE:** This PR uses the code of PR #3726

### Changes
This PR fixes the bug where a scheduled Aggregate Report would always be generated with the name "Concatenated Report". 

I've moved the "save_report" function out of SaveAggregateReportMixin and called it "save_aggregate_report_data". This way, it's possible to access the data from outside of the class. 

I've given the SaveAggregateReportView a "report_type = AggregateOrganisationReport" and I've given the ReportRecipe model a "parent_report_type". This parent_report_type can be "concatinated-report", "aggregated-report" or None (for a single report). 

Then in the ReportRunner it checks if the parent_report_type of the ReportRecipe is an Aggregate Report. If it is, it does a save_aggregate_report_data, else it does a save_report_data (for single or concatenated reports).

### Issue link
Closes #3725 

### Demo
These schedules:
![afbeelding](https://github.com/user-attachments/assets/157bea07-a468-4bbd-978b-2cb082324bec)

Result in:
![afbeelding](https://github.com/user-attachments/assets/e2d0f9dd-1579-4a64-92d0-602ac8b81339)


### QA notes
- Create a scheduled Aggregate Report with "{report type}" in the name
- Change the cron to * * * * * (go to the Scheduled Reports page, open the row, click on the edit button)
- Go back to the Reports History page and refresh until the report is generated
- You should see "Aggregate Report" in the name

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
